File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int count_letters(string text);

int main(void)
{
    string text = get_string("Text: ");

    count_letters(text);
    
    printf("%s\n", text);
}

int count_letters(string text)
{
    int total_letters = 0;
    int length = strlen(text);

    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            total_letters = total_letters + 1;
        }
    }
    return total_letters;
}

File: CS50x/sentimental-readability/readability.py
# TODO

from cs50 import get_string
import re


def main():
    text = get_string("Text: ")

    letters = 0

    words = len(text.split(" "))

    sentences = len(re.split(r"\!|\.|\?", text)) - 1

    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1

    index = round(
        0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8
    )

    if index < 1:
        print("Before Grade 1 ")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Grade ", index)


main()
